tfexec/taint.go

```diff
@@ -0,0 +1,46 @@
+package tfexec
+
+import (
+	"context"
+	"os/exec"
+)
+
+type taintConfig struct {
+	state string
+}
+
+var defaultTaintOptions = taintConfig{}
+
+// TaintOption represents options used in the Taint method.
+type TaintOption interface {
+	configureTaint(*taintConfig)
+}
+
+func (opt *StateOption) configureTaint(conf *taintConfig) {
+	conf.state = opt.path
+}
+
+// Taint represents the terraform taint subcommand.
+func (tf *Terraform) Taint(ctx context.Context, address string, opts ...TaintOption) error {
+	taintCmd := tf.taintCmd(ctx, address, opts...)
+	return tf.runTerraformCmd(ctx, taintCmd)
+}
+
+func (tf *Terraform) taintCmd(ctx context.Context, address string, opts ...TaintOption) *exec.Cmd {
+	c := defaultTaintOptions
+
+	for _, o := range opts {
+		o.configureTaint(&c)
+	}
+
+	args := []string{"taint", "-allow-missing", "-lock-timeout=0s", "-lock=true"}
```

Comment:
```suggestion
	args := []string{"taint", "-no-color"}
```
Given that our intentions behind tfexec is to make it a relatively universal library supporting different use cases, I don't think we should be making any assumptions about the flags here and rely on CLI defaults instead. The only flag which would be basically universally desirable in any automation is `no-color`.

It would be reasonable to introduce each flag as `TaintOption` though, along with `-ignore-remote-version`. Do you mind adding these?